doc/page-implementation.h

```diff
@@ -0,0 +1,80 @@
+/**
+
+\page "Implementation and Usage"
+
+The ordinary implementation an usage of a Facelift interface is described in the Example.
+Here we focus on peculiarities of some selected types.
+
+\section Models
+
+QFace models need to be implemented differently compared to basic properties.
+
+In the following we'll consider this QFace definition:
+\code
+module facelift.imodel 1.0;
+
+interface IModel {
+    readonly model<string> theModel;
+}
+\endcode
+
+We use a model of string types here only to be concise, in real world applications the type is
+usually a custom struct.
+
+\note Only "readonly" models are supported by Facelift currently.
+
+
+\subsection impl-sub1 Implementation
+
+Models can only be implemented in C++, they cannot be implemented in QML. As usual, the
+implementation should derive from IModelPropertyAdapter. Here is the outline of a sample
+implementation:
+
+\code
+class IModelImplementation : public IModelPropertyAdapter
+{
+    Q_OBJECT
+public:
+    IModelImplementation(QObject *parent = nullptr) : IModelPropertyAdapter(parent)
+    {
+        m_theModel.setSize(m_items.size());
+        m_theModel.setGetter(std::bind(&IModelImplementation::getItem, this, std::placeholders::_1));
+    }
+
+    QString getItem(int index)
+    {
+        return m_items.at(index);
+    }
+
+private:
+    QStrigList m_items;
+};
+\endcode
+
+The property adapter will include a public member variable that can be used to provide the model
+data. In our case IModelPropertyAdapter includes \c m_theModel. Two methods of this member variable
+should be called (usually in the constructor): \c setSize and \c setGetter. \c setSize needs to be
```

Comment:
it sounds like setSize would have to be called somewhere by the user code, which is not the case since it is called only by Facelift itself